lib/indent-parser.ts

```diff
@@ -0,0 +1,88 @@
+import * as _ from 'lodash';
+
+export {
+  parse,
+};
+
+// TODO: sort out consistent interface usage
+export interface ParsedTree {
+  [dep: string]: {
+    'NUGET': object;
+  };
+}
+
+// possibly unnecessary repeated parsing without
+// taking advantage of local knowledge of paket.lock particulars
+const traverse = (node: Node, json: any) => {
+  if (node.children.length === 0) {
+    json[node.data] = null;
+  } else {
+    json[node.data] = {};
+    for (const child of node.children) {
+      traverse(child, json[node.data]);
+    }
+  }
+  return json;
+};
+
+class Node {
+  public parent: Node | null;
+  public data: string;
+  public depth: number;
+  public children: Node [];
+
+  constructor(data: any, depth: any) {
+    this.parent = null;
+    this.data = data;
+    this.depth = depth;
+    this.children = [];
+  }
+
+  public toJSON() {
+    return traverse(this, {});
+  }
+}
+
+// parse space indented lines into json format
+function parse(
+  input: string,
+  indent: string = '  ' /* two spaces */,
+  lineSeparator: string = '\n'): ParsedTree | null {
```

Comment:
In original [source code](https://github.com/fsprojects/Paket/blob/master/src/Paket.Core/PaketConfigFiles/DependenciesFile.fs#L811) they replace `\r\n` to `\n` first. Maybe it make sense because this files created on windows machine.